File: app/java/src/main/java/com/example/production/domain/model/process/WorkOrderStatus.java
package com.example.production.domain.model.process;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public enum WorkOrderStatus {
    NOT_STARTED("未着手"),
    IN_PROGRESS("作業中"),
    COMPLETED("完了"),
    SUSPENDED("中断");

    private final String displayName;

    public static WorkOrderStatus fromDisplayName(String displayName) {
        for (WorkOrderStatus status : values()) {
            if (status.displayName.equals(displayName)) {
                return status;
            }
        }
        throw new IllegalArgumentException("Unknown work order status: " + displayName);
    }
}

File: app/csharp/src/ProductionManagement.Domain/Models/Process/WorkOrderStatus.cs
namespace ProductionManagement.Domain.Models.Process;

/// <summary>
/// 作業指示ステータス
/// </summary>
public enum WorkOrderStatus
{
    NotStarted,
    InProgress,
    Completed,
    Suspended
}

public static class WorkOrderStatusExtensions
{
    private static readonly Dictionary<WorkOrderStatus, string> DisplayNames = new()
    {
        { WorkOrderStatus.NotStarted, "未着手" },
        { WorkOrderStatus.InProgress, "作業中" },
        { WorkOrderStatus.Completed, "完了" },
        { WorkOrderStatus.Suspended, "中断" }
    };

    public static string GetDisplayName(this WorkOrderStatus status)
        => DisplayNames[status];

    public static WorkOrderStatus FromDisplayName(string displayName)
        => DisplayNames.First(x => x.Value == displayName).Key;
}
